src/intersections/boxes.jl

```diff
@@ -34,3 +34,71 @@ function intersecttype(f::Function, b1::Box{Dim,T}, b2::Box{Dim,T}) where {Dim,T
     return FaceTouchingBoxes(Box(u, v)) |> f
   end
 end
+
+"""
+    intersecttype(function, ray, box)
+
+Calculate the intersection type of a `ray`` and a `box`` and apply function `f` to it.
+
+## References
+
+* Williams A, Barrus S, Morley R K, et al., 2005. [An efficient and robust ray-box
+  intersection algorithm]
+  (https://dl.acm.org/doi/abs/10.1145/1198555.1198748)
+"""
+function intersecttype(f::Function, r::Ray{3,T}, b::Box{3,T}) where {T}
+  invdir = one(T) ./ direction(r)
+  bounds = coordinates.(extrema(b))
+  orig = coordinates(origin(r))
+
+  # X slab
+  if invdir[1] ≥ 0
+    tmin = (bounds[1][1] - orig[1]) * invdir[1]
+    tmax = (bounds[2][1] - orig[1]) * invdir[1]
+  else
+    tmin = (bounds[2][1] - orig[1]) * invdir[1]
+    tmax = (bounds[1][1] - orig[1]) * invdir[1]
+  end
+  tmin = max.(tmin, zero(T))
+
+  # Y slab
+  if invdir[2] ≥ 0
+    tymin = (bounds[1][2] - orig[2]) * invdir[2]
+    tymax = (bounds[2][2] - orig[2]) * invdir[2]
+  else
+    tymin = (bounds[2][2] - orig[2]) * invdir[2]
+    tymax = (bounds[1][2] - orig[2]) * invdir[2]
+  end
+  if tmin > tymax || tymin > tmax
```

Comment:
So you called the x variable `tmin` and the y variable `tymin`? See the inconsistency? It makes reviewing code a lot harder. Can you please follow the same naming convention for x and y?

Reply:
@juliohm Here `tmin` is the maximum of `txmin`, `tymin` and `tzmin`. It is not similar with `tymin`. I use `tmin` because it covers all 3 dimensions. One in X coordination is calculated first. There is nothing before that to be compared. It is need not to write `tmin = txmin = ...`; only `tmin = ...` is enough. So here is `tmin` instead of `txmin`.

Reply:
In any case, we need to make this logic more explicit by introducing variable names that help the reader of the code understand the logic. I understand there is a variable that you compute in each dimension and a global variable that you update as you process the dimensions. These 3 variables + 1 global variable could be more explicit. You can initialize the global variable with dummy values such as `-Inf, Inf` and then update based on the results per dimension.